File: 900/1850D/main.cpp
#include <bits/stdc++.h>
using namespace std;

int main() 
{
    int t;
    cin >> t;

    while (t--) 
    {
        int n, k;
        cin >> n >> k;

        vector<int> a(n);
        for (int i = 0; i < n; i++)
            cin >> a[i];

        sort(a.begin(), a.end());

        int cnt = 1, ans = 1;
        for (int i = 0; i < n - 1; i++) 
        {
            if (a[i + 1] - a[i] <= k)
                cnt++;
            else
                cnt = 1;
        
            ans = max(ans, cnt);
        }

        cout << (n - ans) << "\n";
    }

    return 0;
}

File: 900/1850D/main.java
import java.util.*;
public class main
{
    public static void main(String args[])
    {
        Scanner sc = new Scanner(System.in);
        
        int t = sc.nextInt();
        
        while(t-- > 0)
        {
            int n = sc.nextInt();
            int k = sc.nextInt();
            
            ArrayList<Integer> a = new ArrayList<>();
            
            for(int i=0;i<n;i++)
                a.add(sc.nextInt());
            
            Collections.sort(a);
            
            int cnt = 1;
            int ans = 1;
            
            for(int i=0;i<n-1;i++)
            {
                if(a.get(i+1)-a.get(i) <= k)
                    cnt++;
                else
                    cnt = 1;
                    
                ans = Math.max(cnt,ans);
            }
            
            System.out.println(n-ans);
        }
    }
}
